### PR TITLE
Fix broken `cylc help license` command

### PR DESCRIPTION
**Pull request by MetRonnie**
Fixes
```
$ cylc help license
Traceback (most recent call last):
  File "~/.conda/envs/tmp/bin/cylc", line 8, in <module>
    sys.exit(main())
  File "~/.conda/envs/tmp/lib/python3.10/site-packages/cylc/flow/scripts/cylc.py", line 626, in main
    print_license()
  File "~/.conda/envs/tmp/lib/python3.10/site-packages/cylc/flow/scripts/cylc.py", line 396, in print_license
    pkg_resources.get_distribution('cylc-flow').get_resource_string(
  File "~/.conda/envs/tmp/lib/python3.10/site-packages/pkg_resources/__init__.py", line 1456, in get_resource_string
    return self._get(self._fn(self.module_path, resource_name))
  File "~/.conda/envs/tmp/lib/python3.10/site-packages/pkg_resources/__init__.py", line 1684, in _get
    with open(path, 'rb') as stream:
FileNotFoundError: [Errno 2] No such file or directory: '~/.conda/envs/tmp/lib/python3.10/site-packages/COPYING'
```

You can do (in a throwaway env)
```
pip install git+https://github.com/MetRonnie/cylc-flow.git@license
```
and `cylc help license` to test this

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are sort of already present (we're testing with an editable install (needed for coverage/Codecov), it's a bit of faff to test with non-editable install).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] No docs change needed
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -74,6 +74,7 @@ install_requires =
     pyzmq==22.*
     # https://github.com/pypa/setuptools/issues/3802
     setuptools>=49, <67
+    importlib_metadata
```
Comment (oliver-sanders) [inline setup.cfg:77]: Looks like this library was added in Py 3.8, can we omit this dependency for 3.8 and above?

```suggestion
    importlib_metadata; python_version < "3.8"
```

File: conda-environment.yml [hunk 2]
```diff
@@ -17,9 +17,9 @@ dependencies:
   - python
   - pyzmq >=22,<23
   - setuptools >=49, <67
+  - importlib_metadata # [py<3.8]
   - urwid >=2,<3
-  # Add # [py<3.11] for tomli once Python 3.11 Released
-  - tomli >=2
+  - tomli >=2 # [py<3.11]
```
Comment (MetRonnie) [inline conda-environment.yml:23]: I am not convinced these comments actually do anything. I can't find any reference to an equivalent to setuptool's environment markers for conda environment YML files online.

Actually, what is this file used for? Is it just CI?

Comment (oliver-sanders) [reply conda-environment.yml:23]: > I am not convinced these comments actually do anything. I can't find any reference to an equivalent to setuptool's environment markers for conda environment YML files online.

https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#preprocessing-selectors

> Actually, what is this file used for? Is it just CI?

This file mirrors the setup.cfg with Conda, it currently has two uses:

* Defining the environment for remote tests.
* Updating the conda-forge feedstock.

We should do more testing with this environment.

**Comment by MetRonnie** [comment]
For some reason the manylinux tests are queueing for hours but not running

**Comment by wxtim** [comment]
Suspect that it's becuase the commit on top of the pile has a "[skip-ci]" in it: I've started my own tests on a copy of the branch https://github.com/wxtim/cylc/actions/runs/4765396470, having deleted that part of the commit message

**Review by wxtim — APPROVED** [review]
Ran the tests on my own branches and all looks well.

**Comment by MetRonnie** [comment]
Oh, it's because Ubuntu 18 is no longer available https://github.com/actions/runner-images/#available-images

**Review by hjoliver — APPROVED** [review]
LGTM, tested.

**Comment by hjoliver** [comment]
(So far as I can tell, no outstanding questions above).

**Pull request merged**